zerver/lib/actions.py

```diff
@@ -5561,7 +5561,19 @@ def do_update_message(
         event[TOPIC_LINKS] = topic_links(message.sender.realm_id, topic_name)
         edit_history_event[LEGACY_PREV_TOPIC] = orig_topic_name
 
+    # Only add content edit history to the message which was
+    # edited, rest of the messages only get topic / stream edit
+    # history.
     update_edit_history(message, timestamp, edit_history_event)
+    if content is not None:
+        content_edit_event_keys = [
+            "prev_content",
+            "prev_rendered_content",
+            "prev_rendered_content_version",
+        ]
+        for key in content_edit_event_keys:
+            del edit_history_event[key]
```

Comment:
I have an even better idea using an added variable name; will just do it and merge.

Reply:
(Also moved this block inside the if statement).